react/hooks/useFetchMore.js

```diff
@@ -157,23 +187,19 @@ export const useFetchMore = props => {
     )
     //if error, rollback
     if (promiseResult && updateQueryError.current) {
-      setCurrentTo(currentTo)
-      setCurrentPage(currentPage)
-      setNextPage(nextPage)
-      setQuery({ page: currentPage }, { replace: true })
+      pageDispatch({ type: 'ROLLBACK' })
```

Comment:
this is weird, did you test it?

Reply:
Yes! This works the same way as `useState`, meaning the values only actually change after the render, so since the rollback is done before the render, this just restores the state to be as it used to be!

Reply:
hmmm. I think this is trusting too much in implementation detail. If I would do it myself, I would pass the args in to rollback in this event as well, to be more declarative. I leave this to the team to decide what is better for their codebase.